scripts/publish/test_publish.py

```diff
@@ -333,6 +351,7 @@ def wait_for_index(
             text=True,
             input=f"{project_name}",
             stdout=PIPE,
+            env=env,
```

Comment:
Noting: I added `env` to `wait_for_index` since we need to propagate the pyx API token in order for `uv pip compile`'s index accesses to succeed.